rfcs/0043-node-error-collection.md

```diff
@@ -0,0 +1,82 @@
+## Summary
+
+[summary]: #summary
+
+This RFC proposes a system that automates the collection of errors from Mina block producer nodes. This system would send an error report when
+1. an exception occurs;
+2. a fatal error occurs.
+By default those error report would be sent to o1labs google cloud backend to facilitate prioritization of fixes for improving the stability of nodes.
+
+## Motivation
+
+[motivation]: #motivation
+
+This system would help o1labs to identify the severity and the frequency of errors. This would greatly facitate the discovery and analysis of the bugs.
+
+## Protocol between the node error collection service and o1labs backend
+
+[protocol]: #protocol
+
+1. Node submits a payload to the backend using `POST /submit` whenever an exception or a fatal error occurs
+2. Server saves the request
+3. Server replies with `200` with `{"status": "OK"}` if the payload is valid
+
+## Interface design
+
+[interface-design]: #interface-design
+
+The node error collection system would make the following `post` request to the designated url.
+
+* `POST /submit` to submit a json payload containing the following data:
+
+```
+{ "data":
+  { "peer_id": "<base64 encoding of the libp2p peer id>"
+  , "ip_address": "<hash of ip address of the submitter>"
+  , "public_key": "<optional, public key of the block producer>"
+  , "git_branch": "<git branch of the mina node>"
+  , "commit_hash": "<commit hash of the mina node>"
+  , "timestamp": "<current time and date>"
+  , "level": "<log level>"
+  , "id": "<random UUID v4>"
+  , "error": "<error message>"
+  , "stacktrace": "<stack trace of the error>"
+  , "cpu": "<cpu info via `lscpu`>"
```

Comment:
Could we leverage `client export-local-logs` to get the hardware information. Maybe attach the tar.gz file to the payload?

Reply:
We could. I don't have a strong opinion on that. I was thinking just fill the field with the return value of `lscpu`.